examples/vision/attention_mil_classification.py

```diff
@@ -0,0 +1,589 @@
+
+    else:
+        print(f"There is no class as {bag_class}")
+        return
+
+    for i in range(PLOT_SIZE):
+        figure = plt.figure(figsize=(8, 8))
+        print(f"Bag number: {labels[i]}")
+        for j in range(BAG_SIZE):
+            image = bags[j][i]
+            figure.add_subplot(1, BAG_SIZE, j + 1)
+            plt.grid(False)
+            if attention_weights is not None:
+                plt.title(np.around(attention_weights[labels[i]][j], 2))
+            plt.imshow(image)
+        plt.show()
+
+
+# Plot some of validation data bags per class.
+plot(val_data, val_labels, "positive")
+plot(val_data, val_labels, "negative")
+
+"""
+## Create model
+
+First we will create some embeddings per instance, invoke the attention operator and then
+use the softmax function to output the class probabilities.
+"""
+
+
+def create_model(instance_shape):
+
+    # Extract features from inputs.
+    inputs, embeddings = [], []
+    for _ in range(BAG_SIZE):
+        inp = layers.Input(instance_shape)
+        flatten = layers.Flatten()(inp)
+        dense_1 = layers.Dense(128, activation="relu")(flatten)
+        dense_2 = layers.Dense(64, activation="relu")(dense_1)
+        inputs.append(inp)
+        embeddings.append(dense_2)
+
+    # Invoke the attention layer.
+    alpha = MILAttentionLayer(
+        weight_params_dim=256,
+        kernel_regularizer=keras.regularizers.l2(0.01),
+        use_gated=True,
+        name="alpha",
+    )(embeddings)
+
+    # Multiply attention weights with the input layers.
+    multiply_layers = [
+        layers.multiply([alpha[i], embeddings[i]]) for i in range(len(alpha))
+    ]
+
+    # Concatenate layers.
+    concat = layers.concatenate(multiply_layers, axis=1)
+
+    # Classification output node.
+    output = layers.Dense(2, activation="softmax")(concat)
+
+    return keras.Model(inputs, output)
+
+
+"""
+## Class weights
+
+Since this kind of problem could simply turn into imbalanced data classification problem,
+class weights should be considered.
+
+Let's say there are 1000 bags. There often could be cases were ~90 % of the bags does not
+contain any positive label and ~10 % does.
+Such data can be referred to as **Imbalanced data**.
+
+Using class weights, the model will tend to consider the rare class more as compared to
+that of the abundant one.
+"""
+
+
+def compute_class_weights(labels):
+
+    # Count number of postive and negative bags.
+    negative_count = len(np.where(labels == 0)[0])
+    positive_count = len(np.where(labels == 1)[0])
+    total_count = negative_count + positive_count
+
+    # Build class weight dictionary.
+    return {
+        0: (1 / negative_count) * (total_count / 2),
+        1: (1 / positive_count) * (total_count / 2),
+    }
+
+
+"""
+## Build and train model
+
+The model is built and trained in this section.
+
+Some regularization techniques are considered to avoid overfitting the model which ensures
```

Comment:
What regularization are you talking about here?

Reply:
I see the regularization in the Create Model block. But seems weird to comment on it here, where you are not actually doing anything related to regularization.

Reply:
Since in this section I am using "Early Stopping" technique, I mentioned regularization techniques here. However, I moved it to the Create Model block since there I'm using regularization as well.